text/0029-block-streaming.md

```diff
@@ -0,0 +1,239 @@
+* Feature Name: `block_streaming`
+* Start Date: 2022-03-01
+* MCIP PR: [mobilecoinfoundation/mcips #29](https://github.com/mobilecoinfoundation/mcips/pull/29)
+* Tracking Issue: [mobilecoin #1433](https://github.com/mobilecoinfoundation/mobilecoin/issues/1433)
+
+# Summary
+[summary]: #summary
+
+This MCIP formalizes the API for getting blocks from consensus nodes.
+
+This includes defining an API for streaming blocks using a dedicated fanout
+mechanism, rather than repeatedly polling S3 for new blocks.
+
+# Motivation
+[motivation]: #motivation
+
+About 40% of the end-to-end transaction time is spent writing to and reading
+from S3. This can be optimized using a dedicated
+[publish/subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern)
+mechanism to distribute updates across nodes.
+
+<details>
+<summary>Details on the 40% figure</summary>
+
+As shown in the screenshot below (dated 2021-11-17), of the 5 seconds to
+externalize a transaction, we spent almost one second uploading to S3, and
+another second downloading from S3.
+
+![trace screenshot](../images/29-trace.png)
+</details>
+
```

Comment:
Can you add one more paragraph to close out the motivation for the proposed change, e.g.

> By using pub/sub block streaming, we will no longer need to poll S3 on regular intervals, and the block download times will greatly decrease.